### PR TITLE
Replace generated AssemblyInfo files with CommonAssemblyInfo

### LINKED ISSUE #133: .NET Core - remove AssemblyInfo generation
In our MSBuild scripts we currently generate AssemblyInfo.cs files and they are ignored in Git. Let's get rid of the generation completely and bolster up CommonAssemblyInfo.cs with conditional compilation to do our bidding.

### PR DESCRIPTION
**Pull request by jonorossi**
Fixes #133.

### CODE REVIEW

File: buildscripts/CommonAssemblyInfo.cs [hunk 1]
```diff
@@ -12,12 +12,41 @@
...
 [assembly: AssemblyCompany("Castle Project")]
-[assembly: AssemblyCopyright("Copyright (c) 2004-2014 Castle Project - http://www.castleproject.org")]
-#if ASSEMBLY_CLSCOMPLIANT_FALSE
-[assembly: System.CLSCompliant(false)]
-#else
-[assembly: System.CLSCompliant(true)]
+[assembly: AssemblyCopyright("Copyright (c) 2004-2016 Castle Project - http://www.castleproject.org")]
+
+[assembly: AssemblyVersion("3.3.0")]
+[assembly: AssemblyFileVersion("3.3.4")]
```
Comment (blairconrad) [inline buildscripts/CommonAssemblyInfo.cs]: Looks good. For ease of future maintenance, I'd consider

1. pulling version numbers out of Settings.proj and having the Get-NuGetVariables read from this file, and
2. maybe making the major/minor/update components into consts so they can be defined once

These things are definitely "bonus".

One thing we do lose, though, is the BuildName in the `AssemblyInformationalVersion`. Comes from TeamCity's Build_Number, I think. This was one of the things that I wasn't sure whether it was important to you.

File: buildscripts/CommonAssemblyInfo.cs [hunk 2]
```diff
@@ -12,12 +12,41 @@
...
+	#if DOTNET45 // order is important
+	"Castle Core for .NET Framework 4.5"
+	#elif DOTNET40
+	"Castle Core for .NET Framework 4.0"
+	#elif DOTNET35
+	"Castle Core for .NET Framework 3.5"
+	#else
+	"Castle Core for .NET Core 1.0"
+	#endif
+)]
```
Comment (blairconrad) [inline buildscripts/CommonAssemblyInfo.cs]: All DLLs will have same name, including the logging plugins.

(Also no "compact framework" for 4.0, which doesn't worry me a bit.)

**Comment by blairconrad** [comment]
Generally looks good me, @jonorossi. The generation (and the output) is much simplified. I'm sorry that my dithering forced _you_ to put more time into it, but I was unsure which of the differing bits were important. :disappointed:

**Comment by blairconrad** [comment]
Oh, is the version number still required in caslte.core/project.json?

File: buildscripts/CommonAssemblyInfo.cs (hunk 1 above)
Comment (jonorossi) [inline buildscripts/CommonAssemblyInfo.cs]: Logged #152 to sort out the consolidation.

File: buildscripts/CommonAssemblyInfo.cs (hunk 2 above)
Comment (jonorossi) [inline buildscripts/CommonAssemblyInfo.cs]: Yep, one of the reasons I got rid of the dot between "Castle" and "Core", I'm setting `AssemblyProduct` and leaving `AssemblyTitle` empty. Being pragmatic the version resource is less important to me than the .NET module name in the assembly, and the file name.

Yep, don't really care that it is sort of meant to be for the client profile, client profile is so dead by now anyway.

**Comment by jonorossi** [comment]
@blairconrad don't worry, I didn't tag this one up-for-grabs because I knew it was going to be a bit hairy especially that we also had a pile of inconsistent `SecurityAssemblyInfo.cs` files littered around too.

I'll merge this then and we can sort of getting consolidated version numbers in #152.

**Pull request merged**